source/AMDSBufferGroup.cpp

```diff
@@ -67,11 +73,28 @@ void AMDSBufferGroup::populateData(AMDSClientStartTimePlusCountDataRequest* clie
 			AMDSFlatArray oneFlatArray;
 			dataHolder->data(&oneFlatArray);
 			clientStartTimePlusCountDataRequest->appendData(dataHolder);
-//			request->histogramData()->append(dataHolders_[iCurrent]);
 		}
 	}
 }
 
+void AMDSBufferGroup::populateData(AMDSClientRelativeCountPlusCountDataRequest* clientDataRequest)
+{
+	int relativeCount = clientDataRequest->relativeCount();
+	int count = clientDataRequest->count();
+
+	int startIndex = dataHolders_.count() - 1 - relativeCount;
+	int endIndex = dataHolders_.count() - 1 - relativeCount + count;
+
+	startIndex = qMax(0, startIndex);
+	endIndex = qMin(endIndex, dataHolders_.count() - 1);
+
+	for (int iCurrent = startIndex; iCurrent < endIndex; iCurrent++)
```

Comment:
Is it possible to index out of bounds with these loop conditions?  How big can `count` be compared to `dataHolders_.size();`?